lib/xcore.js

```diff
@@ -31,14 +31,20 @@ function XClient(displayNum, screenNum, options)
 
     this.displayNum = displayNum;
     this.screenNum = screenNum;
-    this.authHost = os.hostname();
 }
 util.inherits(XClient, EventEmitter);
 
 XClient.prototype.init = function(stream)
 {
     this.stream = stream;
 
+    this.authHost = stream.remoteAddress;
+    this.authFamily = stream.remoteFamily;
+    if (!this.authHost || this.authHost === '127.0.0.1' || this.authHost === '::1') {
+      this.authHost = os.hostname();
+      this.authFamily = null;
+    }
```

Comment:
nothing that I'm aware of. Googled again, some findings for the record:

http://linux.die.net/man/1/xauth
http://stackoverflow.com/questions/28597471/parsing-x-server-authorization-file
https://en.wikipedia.org/wiki/X_Window_authorization

The file format idself is pretty simple but I have not found a good summary for semantics of various auth methods such as MIT-MAGIC-COOKIE-1 XDM-AUTHORIZATION-1 etc